apps/src/sites/hourofcode.com/pages/public/events/index.js

```diff
@@ -257,8 +257,14 @@ function validateFields() {
     } else {
       $('#school-name-error').hide();
     }
+  }
```

Comment:
I split this check for an empty school field and the check for an empty location field up into two separate `if` blocks since the school field previously took precedence even if the school field was hidden at the time.

For example, if the user selected 'Organization/Company' as their event type (which would not require a school field entry) and had a blank location field, the form would correclty not submit but it wouldn't show the error telling them they had to fill in the location field because it would trigger the school `if` block (even though that field would be hidden at the time) and then return before checking the location `if` block, thus never showing the user which field was required that they needed to fill in.